### ISSUE TITLE
Support non-standard fields in Response

### ISSUE THREAD

**Issue by thebabush** [paritytech/jsonrpsee#1349]
Hello,

I'm interacting with a server that sends additional fields in the response message. The fields contain data I care about, so I can't even simply patch jsonrpsee to ignore them (right now jsonerpsee just asserts due to unknown field).

I'm down to make a PR, but I'd like to know if you could guide me through the most painless way to implement this?

Seems like the easiest way would be to extend Response/Notification/etc to have some kind of `unknowns: Value`, but that would mean dynamic allocations + requiring a feature flag, which is ugly.

Thanks.

**Comment by niklasad1** [comment]
Hey hey,

> I'm interacting with a server that sends additional fields in the response message. The fields contain data I care about, so I can't even simply patch jsonrpsee to ignore them (right now jsonerpsee just asserts due to unknown field).

Sorry about that I changed the deserialization of the responses to a manual implementation but forgot about that, fixed in #1353 

> I'm down to make a PR, but I'd like to know if you could guide me through the most painless way to implement this?

> Seems like the easiest way would be to extend Response/Notification/etc to have some kind of unknowns: Value, but that would mean dynamic allocations + requiring a feature flag, which is ugly.

What we could is to introduce another field  `other: HashMap<String, serde_json::value::RawValue>` where it's possible for folks to access custom stuff but kind of annoying allocate stuff that may not be used. We were discussing  to hide it behind a feature flag but I'm not sure I tend to be reluctant to add more feature flags because it's so tricky to maintain etc.

Thus, I think it's fine to add to add `other: HashMap<String, serde_json::value::RawValue>` to the response type.

**Comment by thebabush** [comment]
Thanks for replying.

> Thus, I think it's fine to add to another field `other: HashMap<String, serde_json::value::RawValue>` to the response type.

Problem is RawValue is not sized. Would `Value` work?

Also, I want to get access to the fields, and `process_single_response` just returns the result. It doesn't seem too easy to change that.

**Comment by niklasad1** [comment]
Yeah, you could use either Box<RawValue> or serde_json::Value for that

**Comment by thebabush** [comment]
Well, here it is: https://github.com/thebabush/jsonrpsee/commit/84130eca514e1450ffee6debf5160a5cde8c016c

However, that hashmap is useless atm.
